integration_test/src/python/topology_test_runner/main.py

```diff
@@ -588,19 +587,18 @@ def _run_single_test(topology_name, topology_conf, test_args, http_server_host_p
       expected_state_result_file_path = \
         test_args.topologies_path + "/" + topology_conf["expectedStateResultRelativePath"]
 
-    logging.info("==== Starting test %s of %s: %s ====",
-      current, len(test_topologies), topology_name)
-
     test_threads.append(Thread(target=_run_single_test, args=(topology_name, topology_conf, test_args,
       http_server_host_port, classpath, update_args, deactivate_args, restart_args,
       topology_args, expected_topo_result_file_path, expected_state_result_file_path)))
 
-    current += 1
-
-  for thread in test_threads:
-    thread.start()
-  for thread in test_threads:
-    thread.join()
+  count = 0
+  while count < len(test_threads):
+    for i in range(count, min(count + int(test_args.max_thread_number), len(test_threads))):
```

Comment:
add a new variable and use it, to avoid calculating it twice. it will be easier to maintain.

to_run =  min(count + int(test_args.max_thread_number), len(test_threads)

Reply:
Added.